src/mesh.c

```diff
@@ -0,0 +1,365 @@
+  "dropped_frames_ttl"                // The number of dropped frames due to TTL
+#define FWDED_FRAMES "fwded_frames"   // The number of forwarded frames
+#define FWDED_UNICAST "fwded_unicast" // The number of unicast forwarded frames
+#define FWDED_MCAST "fwded_mcast" // The number of multicast forwarded frames
+
+/*
+ * The config key strings
+ */
+#define PHY_DEVICE_KEY "PhysicalDevice"
+#define NET_DEVICE_KEY "NetworkDevice"
+
+#define PHY_NAME_LEN 32
+#define NET_NAME_LEN 32
+
+struct mesh_list {
+  char phy_name[PHY_NAME_LEN];
+  char net_name[NET_NAME_LEN];
+  struct mesh_list *next;
+}; /* mesh_list */
+typedef struct mesh_list mesh_list_t;
+
+static mesh_list_t *g_mesh_list;
+
+/*
+ * Private functions
+ */
+static int mesh_list_add(const oconfig_item_t *ci) {
+  mesh_list_t *new_mesh;
+
+  if ((ci->values_num != 1) || (ci->values[0].type != OCONFIG_TYPE_STRING)) {
+    WARNING(PLUGIN_NAME " plugin: `" PHY_DEVICE_KEY "' blocks need exactly"
+                        " one string argument.");
+    return EINVAL;
+  }
+
+  if ((new_mesh = calloc(1, sizeof(mesh_list_t))) == NULL) {
+    ERROR(PLUGIN_NAME " plugin: Couldn't allocate memory.");
+    return ENOMEM;
+  }
+
+  sstrncpy(new_mesh->phy_name, ci->values[0].value.string,
+           sizeof(new_mesh->phy_name));
+
+  for (int i = 0; i < ci->children_num; i++) {
+    const oconfig_item_t *child = ci->children + i;
+
+    if (strcasecmp(NET_DEVICE_KEY, child->key) == 0) {
+      if ((child->values_num != 1) ||
+          (ci->values[0].type != OCONFIG_TYPE_STRING)) {
+        WARNING(PLUGIN_NAME " plugin: `" NET_DEVICE_KEY "' blocks need exactly"
+                            " one string argument.");
+        free(new_mesh);
+        return EINVAL;
+      }
+      sstrncpy(new_mesh->net_name, child->values[0].value.string,
+               sizeof(new_mesh->net_name));
+    } else {
+      WARNING(PLUGIN_NAME " plugin: ignoring unknown option %s", child->key);
+    }
+  }
+
+  new_mesh->next = g_mesh_list;
+  g_mesh_list = new_mesh;
+
+  return 0;
+} /* int mesh_list_add */
+
+static int mesh_config(oconfig_item_t *ci) {
+  int ret = 0;
+
+  for (int i = 0; i < ci->children_num; ++i) {
+    oconfig_item_t *child = ci->children + i;
+
+    if (strcasecmp(PHY_DEVICE_KEY, child->key) == 0) {
+      if ((ret = mesh_list_add(child)) != 0)
+        break;
+    } else {
+      WARNING(PLUGIN_NAME " plugin: ignoring unknown option %s", child->key);
+    }
+  }
+
+  return ret;
+} /* int mesh_config */
+
+static int mesh_submit(const mesh_list_t *m, const char *type, value_t value) {
+  value_t values[1];
+  value_list_t vl = VALUE_LIST_INIT;
+
+  values[0] = value;
+
+  vl.values = values;
+  vl.values_len = 1;
+  sstrncpy(vl.plugin, PLUGIN_NAME, sizeof(vl.plugin));
+  sstrncpy(vl.type_instance, m->net_name, sizeof(vl.type_instance));
+  sstrncpy(vl.type, type, sizeof(vl.type));
+
+  return plugin_dispatch_values(&vl);
+} /* int mesh_submit */
```

Comment:
Plugins using deprecated value lists will be disabled by default and need `--enable-compatibility-mode` option: https://github.com/collectd/collectd/releases/tag/collectd-6.0.0.rc0

Use metric families instead: https://github.com/collectd/collectd/blob/collectd-6.0/src/daemon/metric.h

Example on their use: https://github.com/collectd/collectd/blob/collectd-6.0/src/cpusleep.c

(Benefits are that labels can be added to metrics, common labels can be given as OpenTelemetry resource attributes/labels and better metric types.)

Reply:
Got it.